packages/open-truss/src/signals/index.tsx

```diff
@@ -120,41 +120,40 @@ function signalNameFromObject(signal: object | object[]): string {
   return CryptoJS.SHA256(s).toString()
 }
 
-function createZodShape(signal: object | object[]): {
+type DefaultValue = object | object[] | string | boolean | number
```

Comment:
What about string[] (and number and bool)?

Reply:
I'm guessing we will need these but I don't have a workflow right now to exercise them so I'm going to :ship: this as is and we can update once we have a workflow example to exercise/show off the new functionality.

Reply:
Dynamic signals actually already supports arrays for strings, booleans, and numbers. As part of the [recursion for arrays], it grabs the first element in the array and uses that as the element type, then returns a default value array using that element (e.g. `['']`). I guess for whatever reason the types are happy. Perhaps these return values are conforming to `object[]`?

A couple other thoughts:
- This code only uses the type of the first element. This seemed fine since I believe databases only support 1 type (plus null) for a given column, so for most of our current use cases one type is fine. 
- We may eventually need to support union types for signals, e.g. `string | number`. While this isn't necessary for databases, I can see rest APIs or even display components making use of union types for signals. This probably isn't too hard to do with a bit of string parsing, but just noting that we'll need to add support for this.